tests/e2e/framework/kubernetes.go

```diff
@@ -138,11 +139,27 @@ func (k *KubeInfo) Teardown() error {
 	var err error
 	if k.namespaceCreated {
 		if err = util.DeleteNamespace(k.Namespace); err != nil {
-			glog.Error("Failed to delete namespace")
+			glog.Errorf("Failed to delete namespace %s", k.Namespace)
+			return err
+		}
+
+		// confirm the namespace is deleted as it will cause future creation to fail
+		maxAttempts := 15
+		namespaceDeleted := false
+		for attempts := 1; attempts <= maxAttempts; attempts++ {
+			namespaceDeleted, err = util.NamespaceDeleted(k.Namespace)
+			if namespaceDeleted {
+				break
+			}
+			time.Sleep(time.Duration(attempts) * time.Second)
```

Comment:
BTW, the reason I did this way is it actually took a bit over 1 min to delete the namespace in my env, as that also deletes all associated istio resources in that namespace.